examples/main/main.cpp

```diff
@@ -717,7 +717,8 @@ int main(int argc, char ** argv) {
                 const std::string token_str = common_token_to_piece(ctx, id, params.special);
 
                 // Console/Stream Output
-                LOG("%s", token_str.c_str());
+                fprintf(stdout, "%s", token_str.c_str());
+                fflush(stdout);
```

Comment:
Mixing `printf` and `common/log` is not recommended because they run on 2 different threads and the output can get out of order. Maybe try using `LOGV(-1, ...);` and then instead of `--log-disable` use `--log-verbosity -1`?